lib/vector-config/src/stdlib.rs

```diff
@@ -57,7 +57,7 @@ where
         // Said another way, this allows callers to use `#[configurable(derived)]` on a field of `Option<T>` so long as
         // `T` has a description, and both the optional field and the schema for `T` will get the description... but the
         // description for the optional field can still be overridden independently, etc.
-        T::metadata().convert().as_subschema()
+        T::metadata().convert()
```

Comment:
Similar explanation here as above: `as_subschema` drops custom attributes, which is exactly what we don't want to do here when describing the baseline metadata for `T`/`Option<T>`.